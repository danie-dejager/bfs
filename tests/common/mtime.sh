cd "$TEST"

now=$(epoch_time)

"$XTOUCH" -mt "@$((now - 60 * 60 * 24 * 7))" last_week
"$XTOUCH" -mt "@$((now - 60 * 60 * 49))" two_days_ago
"$XTOUCH" -mt "@$((now - 60 * 60 * 25))" yesterday
"$XTOUCH" -mt "@$((now - 60 * 60))" one_hour_ago
"$XTOUCH" -mt "@$((now))" now
"$XTOUCH" -mt "@$((now + 60 * 60 * 24))" tomorrow

bfs_diff . -mindepth 1 \
    \( -mtime -1 -exec printf -- '-mtime -1: %s\n' {} \; -o -true \) \
    \( -mtime  1 -exec printf -- '-mtime  1: %s\n' {} \; -o -true \) \
    \( -mtime +1 -exec printf -- '-mtime +1: %s\n' {} \; -o -true \)
