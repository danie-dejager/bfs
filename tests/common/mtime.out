-mtime  1: ./yesterday
-mtime +1: ./last_week
-mtime +1: ./two_days_ago
-mtime -1: ./now
-mtime -1: ./one_hour_ago
-mtime -1: ./tomorrow
